x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvPSeriesWeightedSumTests.java

```diff
@@ -31,10 +32,21 @@ public MvPSeriesWeightedSumTests(@Name("TestCase") Supplier<TestCaseSupplier.Tes
     @ParametersFactory
     public static Iterable<Object[]> parameters() {
         List<TestCaseSupplier> cases = new ArrayList<>();
-
         doubles(cases);
 
-        // TODO use parameterSuppliersFromTypedDataWithDefaultChecks instead of parameterSuppliersFromTypedData and fix errors
+        cases = randomizeBytesRefsOffset(cases);
+        cases = anyNullIsNull(
+            cases,
+            (nullPosition, nullValueDataType, original) -> nullValueDataType == DataType.NULL ? DataType.NULL : original.expectedType(),
+            (nullPosition, nullData, original) -> {
+                if (nullData.isForceLiteral()) {
+                    return equalTo("LiteralsEvaluator[lit=null]");
+                }
+                return nullData.type() == DataType.NULL ? equalTo("LiteralsEvaluator[lit=null]") : original;
+            }
+        );
+        cases = errorsForCasesWithoutExamples(cases, (valid, position) -> "double");
+
```

Comment:
I wonder if it would be easier to just not require a literal/foldable as the second parameter? We really don't need a literal there, everything should run as smoothly, and maybe we avoid having weird cases with nulls.

Saying this, as I find it weird that we need such extra testing logics for this function. Two birds with one stone (probably?)

Reply:
I had talked with @machadoum about forcing a literal here because it's the way they run it and they can use the forced-literal-ness in the executor. We can always change it later if folks need it. I agree the force-literal-ness makes this function quite a bit more....

Reply:
I feel like we're adding complexity, just to reduce the options. It's done now, so we can go with it, but I would personally remove the literal forcing before doing much more things